codec-http/src/main/java/io/netty/handler/codec/http/HttpMethod.java

```diff
@@ -173,4 +173,44 @@ public String toString() {
     public int compareTo(HttpMethod o) {
         return name().compareTo(o.name());
     }
+
+    private static final class EnumNameMap<T> {
+        private final EnumNameMap.Node<T>[] values;
+        private final int valuesMask;
+
+        EnumNameMap(EnumNameMap.Node<T>... nodes) {
+            values = (EnumNameMap.Node<T>[]) new EnumNameMap.Node[findNextPositivePowerOfTwo(nodes.length)];
+            valuesMask = values.length - 1;
+            for (EnumNameMap.Node<T> node : nodes) {
+                int i = hashCode(node.key) & valuesMask;
+                if (values[i] != null) {
+                    throw new IllegalArgumentException("index " + i + " collision between values: [" +
+                            values[i].key + ", " + node.key + "]");
+                }
+                values[i] = node;
+            }
+        }
+
+        T get(String name) {
+            EnumNameMap.Node<T> node = values[hashCode(name) & valuesMask];
+            return node == null || !node.key.equals(name) ? null : node.value;
+        }
+
+        private static int hashCode(String name) {
+            // This hash code needs to produce a unique index for each HttpMethod. If new methods are added this
+            // algorithm will need to be adjusted. The goal is to have each enum name's hash value correlate to a unique
+            // index in the values array.
+            return name.hashCode() >>> 6;
```

Comment:
can you add a comment "how this would need to be adjusted" ? Like how does the `>>> 6` play into account here

Reply:
I don't have a generic algorithm to determine an appropriate hash code. The only constraint for this class to function as intended is given all the possible `HttpMethod` values, they must each have a unique entry in the array. If this is not the case the constructor will throw and let you know there is a collision (so it won't break silently). The hash code of each value `>>> 6` just happens to be unique relative to each other modulo `16`.

I'll add some additional comments ... let me know if you think there is anything outside of what is described above that would be helpful.